### PR TITLE
Prevent network DoS attack using Extensible payloads 

### LINKED ISSUE #1934: Prevent network DoS attack using Extensible payloads
Drop oracles from the extensible senders list: neo-project/neo#2433, but also implement the second thing from neo-project/neo#2432, that is don't allow anyone to take more than 10-20% of the pool.

### PR DESCRIPTION
**Pull request by fyrchik**
Close #1934.

### CODE REVIEW

File: pkg/network/extpool/pool.go [hunk 1]
```diff
@@ -40,6 +47,10 @@ func (p *Pool) Add(e *payload.Extensible) (bool, error) {
 	p.lock.Lock()
 	defer p.lock.Unlock()
 
+	if len(p.verified) >= p.capacity {
+		return false, nil
```
Comment (roman-khimov) [inline pkg/network/extpool/pool.go:56]: Wrong. You need to kick the oldest entry out of the pool. Which BTW is what we had with Neo 2 consensus messages pool implementation (pool.HashCache).

**Review by roman-khimov — CHANGES_REQUESTED** [review]

File: pkg/config/application_config.go [hunk 2]
```diff
@@ -27,7 +27,9 @@ type ApplicationConfiguration struct {
 	Relay             bool                    `yaml:"Relay"`
 	RPC               rpc.Config              `yaml:"RPC"`
 	UnlockWallet      Wallet                  `yaml:"UnlockWallet"`
-	Oracle            OracleConfiguration     `yaml:"Oracle"`
-	P2PNotary         P2PNotary               `yaml:"P2PNotary"`
-	StateRoot         StateRoot               `yaml:"StateRoot"`
+	// ExtensiblePoolSize is the maximum size of the extensible payload pool.
+	ExtensiblePoolSize int                 `yaml:"ExtensiblePoolSize"`
```
[the remark this answers is not in the document]
Comment (fyrchik) [reply pkg/config/application_config.go:31]: I had first though about using negative for "unlimited size", but we don't need it probably.

File: pkg/network/extpool/pool.go (hunk 1 above)
Comment (fyrchik) [reply pkg/network/extpool/pool.go:56]: Oh, forgot about that, fixed.

Comment (fyrchik) [reply pkg/network/extpool/pool.go:56]: And I am not sure if we should evict items from the same sender. For DoS mitigation current implementation (don't add) is better IMO.

**Review by AnnaShaleva — APPROVED** [review]

File: pkg/network/server.go [hunk 3]
```diff
@@ -30,11 +30,12 @@ import (
...
 	// peer numbers are arbitrary at the moment.
-	defaultMinPeers         = 5
-	defaultAttemptConnPeers = 20
-	defaultMaxPeers         = 100
-	maxBlockBatch           = 200
-	minPoolCount            = 30
+	defaultMinPeers           = 5
+	defaultAttemptConnPeers   = 20
+	defaultMaxPeers           = 100
+	defaultExtensiblePoolSize = 100
```
Comment (roman-khimov) [inline pkg/network/server.go:36]: It's per-sender now, so 20 makes more sense.

**Review by roman-khimov — APPROVED** [review]

**Pull request merged**